include/xtensor-r/rcpp_extensions.hpp

```diff
@@ -80,9 +80,9 @@ namespace Rcpp
             {
             }
 
-            inline xt::roptional_assembly<T> get()
+            inline xt::rcontainer_optional<T> get()
             {
-                return xt::roptional_assembly<T>(m_sexp);
```

Comment:
I should probably double check but, just like `Exporter` is specialized for both `rarray` and `rtensor`, we should specialize `Exporter` for `rtensor_optional` and `rarray_optional`.

Reply:
Gotcha. I wasn't sure about this since `rarray_optional` and `rtensor_optional` are declared with `using` and I think `rarray` is a class instead.

Reply:
ah, I think that it was fine in fact.

I rebased the PR (upon the move to xtensor 0.20.1), and removed that last commit.